txpool/txpool.go

```diff
@@ -641,3 +679,38 @@ func (t *txPriceHeapImpl) Pop() interface{} {
 	*t = old[0 : n-1]
 	return job
 }
+
+type localAccounts struct {
+	accounts map[types.Address]bool
+	mutex    sync.RWMutex
```

Comment:
Do we really need a mutex if we are only setting the localAccounts when starting up the node, and after that we are only reading from different threads?

Reply:
There is a possibility that accounts are added to this struct after starting. If noLocals is disabled, address is added to local accounts when new local transaction is inserted to txPool. This is used when reorg happens (In IBFT, it won't happen)

Reply:
Got it, thanks for the answers, we shouldn't count on the consensus having instant finality 👍